src/TTextEdit.cpp

```diff
@@ -391,20 +392,26 @@ void TTextEdit::drawLine(QPainter& painter, int lineNumber, int lineOfScreen) co
         TChar timeStampStyle(QColor(200, 150, 0), QColor(22, 22, 22));
         QString timestamp(mpBuffer->timeBuffer.at(lineNumber));
         for (const QChar c : timestamp) {
-            cursor.setX(cursor.x() + drawGrapheme(painter, cursor, c, 0, timeStampStyle));
+            cursor.setX(cursor.x() + drawGrapheme(painter, cursor, c, 0, timeStampStyle).first);
         }
     }
 
     int columnWithOutTimestamp = 0;
+    int spaceForWidestGlyph = 0;
     for (int indexOfChar = 0, total = lineText.size(); indexOfChar < total;) {
         int nextBoundary = boundaryFinder.toNextBoundary();
 
         TChar& charStyle = mpBuffer->buffer.at(lineNumber).at(indexOfChar);
-        int graphemeWidth = drawGrapheme(painter, cursor, lineText.mid(indexOfChar, nextBoundary - indexOfChar), columnWithOutTimestamp, charStyle);
-        cursor.setX(cursor.x() + graphemeWidth);
+        QString grapheme{lineText.mid(indexOfChar, nextBoundary - indexOfChar)};
+        QPair<int, int> graphemeWidths = drawGrapheme(painter, cursor, grapheme, columnWithOutTimestamp, charStyle);
+        cursor.setX(cursor.x() + graphemeWidths.first);
         indexOfChar = nextBoundary;
-        columnWithOutTimestamp += graphemeWidth;
+        columnWithOutTimestamp += graphemeWidths.first;
+        if (grapheme != QChar::Tabulation) {
+            spaceForWidestGlyph = qMax(spaceForWidestGlyph, qRound(0.5 + graphemeWidths.second / static_cast<double>(graphemeWidths.first)));
```

Comment:
```suggestion
            spaceForWidestGlyph = std::max(spaceForWidestGlyph, qRound(0.5 + graphemeWidths.second / static_cast<double>(graphemeWidths.first)));
```

Better to use standard library where we can!

Reply:
We'll probably find it is macro-ed to that anyhow nowadays - but I will change it here and elsewhere.